### PR TITLE
feat(authz): Adds operation checker to ensure operation description o…

### PR DESCRIPTION
**Pull request by ttomsu**
…bjects are implementing Fiat interfaces.

cc: @duftler @danielpeach @owainlewis

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/converters/autoscaler/UpsertKubernetesAutoscalerAtomicOperationConverter.groovy [hunk 1]
```diff
@@ -34,7 +34,7 @@ class UpsertKubernetesAutoscalerAtomicOperationConverter extends AbstractAtomicO
   }
 
   @Override
-  Object convertDescription(Map input) {
+  KubernetesAutoscalerDescription convertDescription(Map input) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/converters/autoscaler/UpsertKubernetesAutoscalerAtomicOperationConverter.groovy]: Thank you :)

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/listeners/OperationsTypeChecker.groovy [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+   */
+  @Override
+  void onApplicationEvent(ContextRefreshedEvent event) {
+    // Docker's annotation is just the Annotation class, which is unhelpful and thus removed.
+    def annotationTypes = (cloudProviders).collect { it.getOperationAnnotationType() } - Annotation
+    log.info("Found ${annotationTypes.size()} cloud provider annotations: ${annotationTypes*.simpleName}")
+
+    annotationTypes.each { annotationType ->
+      def beanMap = applicationContext.getBeansWithAnnotation(annotationType)
+      log.debug("~~~")
```
Comment (lwander) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/listeners/OperationsTypeChecker.groovy]: Intentional?

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/listeners/OperationsTypeChecker.groovy [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      }
+    }
+  }
+
+  static Class getConvertDescriptionReturnType(Object instance) {
+    try {
+      Method convertDescriptionMethod = instance.class.getMethod("convertDescription", Map)
+      return convertDescriptionMethod.returnType
+    } catch (MethodNotFoundException mnfe) {
+      log.debug("Cannot find 'convertDescription' method", mnfe)
```
Comment (lwander) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/listeners/OperationsTypeChecker.groovy]: Should we `return null` to be clear here?

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/listeners/OperationsTypeChecker.groovy (hunk 2 above)
Comment (ttomsu) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/listeners/OperationsTypeChecker.groovy]: Yes - it's meant to be a small callout to whomever is reading the logs

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/listeners/OperationsTypeChecker.groovy (hunk 3 above)
Comment (ttomsu) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/listeners/OperationsTypeChecker.groovy]: Done

**Pull request merged**